src/main/java/recipeio/InputParser.java

```diff
@@ -104,35 +106,36 @@ public static MealCategory parseMealCriteria(String userInput) {
         return mealCategory;
     }
 
-    public static Recipe parseAdd(String userInput) throws Exception{
-        String[] words = userInput.trim().split(" ");
-        // Ignore the first word and join the remaining words into a string
-        //add pizza/34/340/eggs/dinner/www.food.com
-        String[] remainingInput = parseDetails(userInput);
-        assert remainingInput.length > 0;
+    public static Recipe parseAdd(String userInput) throws Exception {
+        String[] words = userInput.trim().split(" ", 2);
+        String[] remainingInput = words[USER_INPUT_INDEX].trim().split(RECIPE_DELIMETER);
+        assert remainingInput.length > 0 : "Add additional parameters to add command";
         checkCorrectAddFormat(remainingInput);
```

Comment:
Your checkCorrectAddFormat function should be a boolean function, as per the rest of the validators in Command Validator.

Do consider shifting your validation code into the CommandValidator class, and name it `isValidAddCommand`. 

Meanwhile, the InputParser is simply there to extract elements from the user's input.